### PR TITLE
fix: optimize the judgment of local cache expiration

### PR DESCRIPTION
**Pull request by lowzj**
Signed-off-by: lowzj <zj3142063@gmail.com>

### Ⅰ. Describe what this PR did

* increase read timeout to 2 seconds
* not expire if get response from remote server timeout

### CODE REVIEW

File: src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java [hunk 1]
```diff
@@ -84,8 +82,9 @@ public static boolean isExpired(String fileUrl, long lastModified, String eTag,
                 conn = openConnection(url);
                 fillHeaders(conn, headers);
                 conn.setUseCaches(false);
+                //conn.setRequestMethod("HEAD");
                 conn.setConnectTimeout(2000);
-                conn.setReadTimeout(1000);
+                conn.setReadTimeout(2000);
```
Comment (Starnop) [inline src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java]: How did you get this value? I think it is unreasonable if it is based on experience. 
Could we make it configurable? 
+ If that, we have to do that without hesitation . 
+ If not, let's take a step back and set a constant.

WDYT?

File: src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java [hunk 2]
```diff
@@ -84,8 +82,9 @@ public static boolean isExpired(String fileUrl, long lastModified, String eTag,
                 conn = openConnection(url);
                 fillHeaders(conn, headers);
                 conn.setUseCaches(false);
+                //conn.setRequestMethod("HEAD");
```
Comment (Starnop) [inline src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java]: Delete it or record a TODO item?

File: src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java [hunk 3]
```diff
@@ -126,6 +128,7 @@ public static long getContentLength(String fileUrl, String[] headers, boolean df
                 conn = openConnection(url);
                 conn.setUseCaches(false);
                 fillHeaders(conn, headers);
+                //conn.setRequestMethod("HEAD");
```
Comment (Starnop) [inline src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java]: So do with above.

File: src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java (hunk 1 above)
Comment (lowzj) [reply src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java]: Keep the same with the value in function `getContentLength`. I have no plan to make it configurable in java-version. Maybe we could do that when we refactor supernode with golang.

Comment (Starnop) [reply src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java]: Then set a constant is better. WDYT?

File: src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java [hunk 4]
```diff
@@ -28,16 +34,8 @@
...
-import javax.net.ssl.HostnameVerifier;
-import javax.net.ssl.HttpsURLConnection;
-import javax.net.ssl.SSLContext;
-import javax.net.ssl.SSLSession;
-import javax.net.ssl.TrustManager;
-import javax.net.ssl.X509TrustManager;
-
 import com.dragonflyoss.dragonfly.supernode.common.exception.AuthenticationRequiredException;
 import com.dragonflyoss.dragonfly.supernode.common.exception.UrlNotReachableException;
-
```
Comment (Starnop) [inline src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java]: Why delete it?

File: src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java (hunk 1 above)
Comment (lowzj) [reply src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java]: I think that this work can be done in supernode go-version, and java-version we just do some bugfix work.

Comment (chenchaobing) [reply src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/common/util/HttpClientUtil.java]: > I think that this work can be done in supernode go-version, and java-version we just do some bugfix work.

agree

**Comment by chenchaobing** [comment]
LGTM

**Pull request merged**